### ISSUE TITLE
"Local variable 'row' referenced before assignment" when running Snowflake profiler ingestion

### ISSUE THREAD

**Issue by aldwyn** [open-metadata/OpenMetadata#14286]
**Affected module**
Ingestion Framework

**Describe the bug**
```
DIM_PERSONAL_INFO.BIRTH_YEARetric_type.value: local variable 'row' referenced before assignment
```

**To Reproduce**
Create a simple profiler ingestion in the Snowflake service with the default parameters.

**Expected behavior**
The profiler ingestion should pass through the `compute_metrics_in_thread` function and return the `row` values successfully.

**Version:**
 - OS: Windows 10 Enterprise
 - Python version: 3.9
 - OpenMetadata version: 1.2.2
 - OpenMetadata Ingestion package version: `openmetadata-managed-apis==1.2.2`, `openmetadata-ingestion[snowflake,dbt,powerbi,pii-processor,postgres]==1.2.2`

**Additional context**
To fix this, the `row = None` in the `except` statement must be transposed to before the `try` statement, e.g.:
```python
row = None
try:
    row = <some_value>
except:
    <some_code>
return row
```

**Comment by aldwyn** [comment]
#14287

**Comment by harshach** [comment]
@TeddyCr can we fix this for 1.2.4?
